web/packages/teleport/src/Account/ManageDevices/AddAuthDeviceWizard/AddAuthDeviceWizard.tsx

```diff
@@ -246,51 +284,160 @@ function CreatePasskeyStep({
   );
 }
 
+function CreateMfaBox({
+  auth2faType,
+  newMfaDeviceType,
+  privilegeToken,
+  onNewMfaDeviceTypeChange,
+}: {
+  auth2faType: Auth2faType;
+  newMfaDeviceType: Auth2faType;
+  privilegeToken: string;
+  onNewMfaDeviceTypeChange(o: Auth2faType): void;
+}) {
+  const mfaOptions = createMfaOptions({
+    auth2faType,
+    required: true,
+  }).map((o: MfaOption) =>
+    // Be more specific about the WebAuthn device type (it's not a passkey).
+    o.value === 'webauthn' ? { ...o, label: 'Hardware Device' } : o
+  );
+
+  return (
+    <>
+      Multi-factor type
+      <RadioGroup
+        name="mfaOption"
+        options={mfaOptions}
+        value={newMfaDeviceType}
+        flexDirection="row"
+        gap={3}
+        onChange={o => {
+          onNewMfaDeviceTypeChange(o as Auth2faType);
+        }}
+      />
+      {newMfaDeviceType === 'otp' && (
+        <QrCodeBox privilegeToken={privilegeToken} />
+      )}
+    </>
+  );
+}
+
+function QrCodeBox({ privilegeToken }: { privilegeToken: string }) {
+  const fetchQrCodeAttempt = useAttempt('');
+  const [qrCode, setQrCode] = useState('');
+
+  useEffect(() => {
+    fetchQrCodeAttempt.run(() =>
+      auth
+        .createMfaRegistrationChallenge(privilegeToken, 'totp')
+        .then(res => setQrCode(res.qrCode))
+    );
+  }, []);
+
+  return (
+    <Flex
+      flexDirection="column"
+      borderRadius={8}
+      p={3}
+      mb={4}
+      css={`
+        background: ${props => props.theme.colors.interactive.tonal.neutral[0]};
+      `}
+    >
+      <Flex height="168px" justifyContent="center" alignItems="center">
+        {fetchQrCodeAttempt.attempt.status === 'processing' && <Indicator />}
+        {fetchQrCodeAttempt.attempt.status === 'success' && (
+          <Image
+            src={`data:image/png;base64,${qrCode}`}
+            height="100%"
+            style={{
+              boxSizing: 'border-box',
+              border: '8px solid white',
+            }}
+          />
+        )}
+      </Flex>
+      <Text typography="body1" textAlign="center" mt={2}>
+        Scan the QR Code with any authenticator app.
+        <br />
+        We recommend{' '}
+        <Link href="https://authy.com/download/" target="_blank">
+          Authy
+        </Link>
+        .
+      </Text>
+    </Flex>
+  );
+}
+
 interface SaveKeyStepProps extends StepComponentProps {
   privilegeToken: string;
   credential: Credential;
+  usage: DeviceUsage;
+  newMfaDeviceType: Auth2faType;
   onSuccess(): void;
 }
 
-function SavePasskeyStep({
+export function SaveDeviceStep({
```

Comment:
```suggestion
export function SaveMfaStep({
```

Perhaps a more generic name? (Since some would argue that TOTP is not a "device")

Reply:
Since the entire wizard is called `AddAuthDeviceWizard`, I'm using an "auth device" as an abstract term for all of these, and MFA is actually a less generic name (since it explicitly excludes passkeys). Note that I inherited the "device" terminology from existing code (`AddDevice.tsx`, `useAddDevice`, etc.). If we want to rename the concept of "device" to something else, perhaps we could consider "auth method", and rename the appropriate elements to `AddAuthMethodWizard`, `SaveMethodStep`, etc. However, as the "device" terminology is also used across our backend codebase, I'd say it's gonna end up being more confusing than useful.

Reply:
I still think `SaveMfaStep` is best.

"Devices" is an overloaded term in Teleport, because we also have _Device Trust_ which keeps an inventory of trusted devices in the backend.

Reply:
OK, I can rename it, but I'd like to know your ideas about renaming `CreateMfaBox`, then. SInce with `SaveMfaStep`, MFA will actually "MFA _or_ a passkey", while in case of `CreateMfaBox`, MFA will mean _exactly_ MFA. Is there any specific name I could then put on "MFA, but _really_ MFA"?

Reply:
(This, plus a bunch of other variables with `mfa` in name and whose names will then be confusing.)